File: 2199-two-furthest-houses-with-different-colors/2199-two-furthest-houses-with-different-colors.java
class Solution {
    public int maxDistance(int[] colors) {
        int maxDistance = 0;
        for(int i = 0; i <colors.length; i++){
            if(colors[0] != colors[i]){
                maxDistance = Math.max(maxDistance, i);
            }
            if(colors[colors.length - 1] != colors[i]){
                maxDistance = Math.max(maxDistance, colors.length - i - 1);
            }
        }
        return maxDistance;
    }
}

File: 2199-two-furthest-houses-with-different-colors/2199-two-furthest-houses-with-different-colors.py
class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        maxDistance = 0
        
        for i in range(1, len(colors)):
            if colors[0] != colors[i]:
                maxDistance = max(maxDistance, i)
            if colors[i] != colors[-1]:
                maxDistance = max(maxDistance, len(colors) - i - 1)
            
        return maxDistance


